scripts/build_peach_go_sbot.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+"""
+script to create debian packages for cross-compiled go binaries for go-sbot
+based off of this post
+https://unix.stackexchange.com/questions/627689/how-to-create-a-debian-package-from-a-bash-script-and-a-systemd-service
+"""
+
+
+from constants import *
+from utils import render_template
+
+import subprocess
+import argparse
+import sys
+import os
+import shutil
+
+DEB_CONF_DIR = os.path.join(PROJECT_PATH, 'conf/templates/peach_go_sbot')
+DEB_BUILD_DIR = "/tmp/peach_go_sbot"
+GO_SSB_DIR = "/srv/peachcloud/automation/go-ssb"
+
+
+def crosscompile_peach_go_sbot():
+    subprocess.check_call(["git", "pull"], cwd=GO_SSB_DIR)
+    # TODO: confirm that version number in the repo matches the version number we set for the package we are building
+    print("[CROSS-COMPILING sbotcli]")
+    subprocess.check_call(["env", "GOOS=linux", "GOARCH=arm64", "go", "build", "./cmd/sbotcli"], cwd=GO_SSB_DIR)
+    print("[CROSS-COMPILING go-sbot]")
+    subprocess.check_call(["env", "GOOS=linux", "GOARCH=arm64", "go", "build", "./cmd/go-sbot"], cwd=GO_SSB_DIR)
+
+def package_peach_go_sbot(version):
+
+    print("[ PACKAGING peach-go-sbot ]")
+    # copy debian conf files into correct locations in package build directory
+    DEBIAN_SRC_DIR = os.path.join(DEB_CONF_DIR, 'DEBIAN')
+    DEBIAN_DEST_DIR = os.path.join(DEB_BUILD_DIR, 'DEBIAN')
+    os.makedirs(DEBIAN_DEST_DIR)
+    maintainer_scripts = ['postinst', 'postrm', 'prerm']
+    for script in maintainer_scripts:
+        src = os.path.join(DEBIAN_SRC_DIR, script)
+        dest = os.path.join(DEBIAN_DEST_DIR, script)
+        shutil.copyfile(src, dest)
+        subprocess.check_call(["chmod", "775", dest])
+    # copy control file putting in correct version number
+    src = os.path.join("peach_go_sbot/DEBIAN/control")
+    dest = os.path.join(DEBIAN_DEST_DIR, "control")
+    render_template(src=src, dest=dest, template_vars={"version": version})
+
+    # copy systemd service file
+    SERVICE_DIR = os.path.join(DEB_BUILD_DIR, 'lib/systemd/system')
```

Comment:
Is `SERVICE_DIR` missing a forward slash?

`DEB_BUILD_DIR` = "/tmp/peach_go_sbot" and we're appending "lib/systemd/system", which would make `SERVICE_DIR` =

`/tmp/peach_go_sbotlib/systemd/system`

Maybe that's intended and I'm missing something?

Reply:
Same thing appears on line 59 for `BIN_DIR`.

Reply:
I just confirmed in a python shell, os.path.join inserts the "/" in between parts as needed.
the way I think about it, the second part is like a relative path joined to the first part (and so the second part doesn't start with a slash)

Reply:
Ok awesome, thanks for checking! I figured it must work somehow, since you've already been using the script successfully. Nice to have learned a little extra about `os.path.join`.